### ISSUE TITLE
Bug 03 Entering position between 0 and 1 degree W gives false position.

### ISSUE THREAD

**Issue by trudK45** [rgleason/weather_routing_pi#99]
If you try to enter a position of say 30 minutes W by entering -0 degrees 30 minutes you will end up with a position of 30 minutes east. It's obvious that -0 is interpreted as 0 and the minus sign is not transferred to the minutes part.

**Comment by trudK45** [comment]
Looking at the code I think this also applies for 0 -1 degrees south.
You simply need to detect the entered minus sign on the degrees to decide if the minutes are to be negated or not if the value of the degrees is zero.

The routine OnNewPosition in file WeatherRouting.cpp:945
void WeatherRouting::OnNewPosition( wxCommandEvent& event )
{
    NewPositionDialog dlg(this);
    if(dlg.ShowModal() == wxID_OK) {
        double lat=0, lon=0, lat_minutes=0, lon_minutes=0;

        wxString latitude_degrees = dlg.m_tLatitudeDegrees->GetValue();
        wxString latitude_minutes = dlg.m_tLatitudeMinutes->GetValue();
        latitude_degrees.ToDouble(&lat);
        latitude_minutes.ToDouble(&lat_minutes);
        lat_minutes = fabs(lat_minutes);
        if(lat < 0) <------------------------------------------------ This will not work if -0 is entered
            lat_minutes = -lat_minutes;
        lat += lat_minutes / 60;

        wxString longitude_degrees = dlg.m_tLongitudeDegrees->GetValue();
        wxString longitude_minutes = dlg.m_tLongitudeMinutes->GetValue();
        longitude_degrees.ToDouble(&lon);
        longitude_minutes.ToDouble(&lon_minutes);
        lon_minutes = fabs(lon_minutes);
        if(lon < 0) <------------------------------------------------ This will not work if -0 is entered
            lon_minutes = -lon_minutes;
        lon += lon_minutes / 60;

        AddPosition(lat, lon, dlg.m_tName->GetValue());
    }
}

**Comment by sebastien-rosset** [comment]
See #221 . I'm proposing to remove the "Positions" panel entirely and use the marks from OpenCPN. I.e. the user should be able to create marks in OpenCPN core and use them directly in the WR plugin, without having to do an intermediate step to convert them to position in the WR plugin.

I'm open to counter-argument that Position are necessary.

**Comment by rgleason** [comment]
This is not about positions really, it is about entering positions between 0 and 1 degree W giving false positions.
It is a bug in the calculations that may not change unless we find it.

**Comment by trudK45** [comment]
@rgleason 
>  It is a bug in the calculations that may not change unless we find it.

The bug is all ready found. As I wrote in my second post :

>if(lat < 0) <------------------------------------------This will not work if -0 is entered
>       lat_minutes = -lat_minutes;
>    lat += lat_minutes / 60;

This does not works as -0 is not less than 0 and the lat/long will be regarded as positive.
You have to take care of the case -0 degrees XX minutes lat/long.

**Comment by sebastien-rosset** [comment]
Can you provide the steps to reproduce? There are multiple ways to enter positions. Are you clicking the menu Positions, then "New Position"? 

<img width="472" alt="Image" src="https://github.com/user-attachments/assets/636a2cca-384d-4f32-b12c-539635f8953d" />

**Comment by rgleason** [comment]
Thanks trudk45  I see your point Sebastien, for closing this.
I wonder if there is any case for manual entry of position?  It certainly is more intuitive to use a cursor or mouse.
In the case of android and fat inaccurate fingers I've found in placing routes that I can drag wp ok, even on my tiny cell phone. I think it would be a bigger pain to try to enter position manually on Android, in fact in my case it would Never happen.

While most users with PC's don't have touch screens yet, but use the mouse, I think it is common for Macs to have touch.   So I think we can close this Issue because it is going to become a "mote" point.  Ok?

Can we close it?

**Comment by trudK45** [comment]
@rgleason
> Can we close it?
In that case I think the option of entering position within WR should be removed.
I myself usually drop a mark in OpenCPN core and make it a WR position.

**Comment by rgleason** [comment]
Yes, I believe that is what Sebastien intends to do, but it is best to have him clarify this and let him close this now.

**Comment by trudK45** [comment]
About "RFE: 04 Interface Improvement: Inconsistency between deleting a position and a configuration":
@sebastien-rosset says "I think the entire "positions" panel should disappear. "

If you do remove the "positions" as suggested in this Issue, how do you remove a position after you have added it ?

**Comment by rgleason** [comment]
Using the normal OpenCPN commands for Routes and WP.  Right click and pick delete.  Or Move WP or whatever.

**Comment by trudK45** [comment]
@rgleason 
> Using the normal OpenCPN commands for Routes and WP. Right click and pick delete. Or Move WP or whatever.

Have you tried this ?

I dropped a mark, made it a WR position. It shows up in WR position window. Then I deleted the mark in OpenCPN core but it  does still present itself in the WR position window.
Same thing with a route where one of the WP's are made into a WR routing position.

**Comment by sebastien-rosset** [comment]
> Using the normal OpenCPN commands for Routes and WP. Right click and pick delete. Or Move WP or whatever.

Yes, that's the idea.

> Have you tried this ?
> I dropped a mark, made it a WR position. It shows up in WR position window. Then I deleted the mark in OpenCPN core but it does still present itself in the WR position window.
Same thing with a route where one of the WP's are made into a WR routing position.

Right, that's because it's not implemented yet.  The WR plugin would discover the marks from OpenCPN and make sure everything is in sync. Any mark created in OpenCPN would be immediately available to create routings. The user will also be able to select a mark and right click "create routing from boat to mark" in a single step.
This will need to come with some additional enhancements, like the ability to start typing characters in the start/end drop down lists, potentially with filters.
If the user deletes a mark which is referenced in a routing, then that would mean the routing would no longer be valid.

We would also align the WR terminology with OpenCPN core. I.e. use "marks", not position.

**Comment by trudK45** [comment]
@rgleason 

>What versions of Opencpn, Climatology and WR are you using?

5.8.0-0 as I am weatherrouting folks across the Atlantic and don't want any surprises.

**Comment by rgleason** [comment]
Oh, very interesting. 
When we get this closer to the finish, We'd really like to you test it in a parallel installation. 
Can you do that?
Also please keep an eye here on all Issues and chime in.
We are probably going change the Configuration Dialog and would really like to have more discussion and thoughts about how to handle it.  See GOAL: Declutter and reduce Configuration Dialog #242

**Comment by rgleason** [comment]
@sebastien-rosset  I'm closing this because we will are eliminating the entry form for Position. Is this Correct?
